File: LaboratoryWorkOOP2.3(sharp)/LaboratoryWorkOOP2.3(C++)/Letter.cpp
#include "stdafx.h"
#include "Letter.h"


Letter::Letter()
{
}


Letter::~Letter()
{
}

Letter::Letter(string usrStr)
{
	for (int i = 0; i < usrStr.size(); i++)
		letters.push_back(usrStr[i]);
}

int Letter::GetSize()
{
	return letters.size();
}

void Letter::Show()
{
	for each (int var in letters)
		cout << var;
	cout << endl;
}

void Letter::DeleteSmth()
{
	for (int i = 0; i < letters.size() - 1; i++)
		if (letters[i] == 'a')
			swap(letters[i], letters[i + 1]);
	letters.pop_back();
	for (int i = 0; i < letters.size() - 1; i++)
		if (letters[i] == 'a')
			DeleteSmth();
	Show();
}

File: LaboratoryWorkOOP2.3(sharp)/LaboratoryWorkOOP2.3(sharp)/Letter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LaboratoryWorkOOP2._3_sharp_
{
    class Letter : Row
    {
        private List<char> letters;

        public Letter(string userLetters)
        {
            letters = new List<char>();
            char[] tmpArr = userLetters.ToCharArray();
            foreach (var elem in tmpArr)
                if (char.IsLetter(elem))
                    letters.Add(elem);
        }

        public override void DeleteSmth()
        {
            for (int i = 0; i < letters.Count; i++)
                letters.Remove('a');
            Show();
        }

        public override int GetSize()
        {
            return letters.Count;
        }

        public override void Show()
        {
            foreach (var elem in letters)
                Console.Write(elem);
            Console.WriteLine();
        }
    }
}
